File: C/ladder1.c
#include <stdio.h>

void move()
{
    

    for(int row = ar; row < 100; row++)
    {
        for(int col = 0; col < 100; col++)
        {

        }
    }
}

int main()
{   
    for(int t = 1; t <= 10; t++)
    {
        //테스트케이스 번호 받기
        int T;
        scanf("%d", &T);

        //사다리 행렬 받기
        int ladder[100][100] = {};

            //도착지점 기록
        int arriveX = 0, arriveY = 0;

        for(int i = 0; i < 100; i++)
        {
            for(int j = 0; j < 100; j++)
            {
                scanf("%d ", &ladder[i][j]);

                if(ladder[i][j] == 2)
                {
                    arriveY = i;
                    arriveX = j;
                }
            }
        }

        //알고리즘
        //방향벡터: 위, 왼, 오
        int dx[3] = {-1, 0, 0};
        int dy[3] = { 0,-1, 1};

        while(1)
        {
            //1순위: 왼쪽
            if()

            //2순위: 오른쪽

            //3순위: 위쪽
        }

        //결과출력
        printf("#%d %d", t, result);
    }

    return 0;
}

File: Python/Ladder1.py
def find_start(data, start_row, start_col):

    #파라미터 초기화
    row = start_row
    col = start_col

    while(0 <= row < 100 and 0 <= col < 100):
        
        #그냥 지나가면 왼쪽1과 오른쪽1이 충돌이 일어남
        #그래서 지나간 자리는 0으로 초기화 해야함
        #우선순위1: 왼쪽(오른쪽)
        if(col-1 >= 0 and data[row][col-1] == 1):
            data[row][col] = 0
            col = col - 1

        #우선순위2: 오른쪽(왼쪽)
        elif(col+1 < 100 and data[row][col+1] == 1):
            data[row][col] = 0
            col = col + 1
        
        #우선순위3: 위쪽
        elif(row-1 >= 0 and data[row-1][col] == 1):
            data[row][col] = 0
            row = row - 1

        if(row == 0):
            return col
        


#총 10개의 테스트케이스가 주어진다
for t in range(1, 11):

    T = int(input())

    #파라미터 초기화
    N = 100
    end_row = 99
    end_col = 0

    #사다리 받기
    ladder = [list(map(int, input().split())) for _ in range(N)]

    #도착지점 찾아놓기
    for j in range(N):
        if(ladder[99][j] == 2):
            end_col = j

    #알고리즘, 결과출력
    print(f'#{t} {find_start(ladder, 99, end_col)}')
    